frontend/src/utils/TaskUtils.ts

```diff
@@ -220,51 +220,64 @@ export const dragDropBetweenLists = (
   };
 };
 
-export const calcTaskWeightedValueSum = (
-  task: Task,
-  allCustomers: Customer[],
-  roadmap: Roadmap,
+const ratingValue = (allCustomers: Customer[], roadmap: Roadmap) => (
+  rating: Taskrating,
 ) => {
-  const customerValuesSum = allCustomers.reduce(
-    (total, customer) => total + customerWeight(customer),
-    0,
+  const ratingCreator = allCustomers.find(
+    ({ id }) => id === rating.forCustomer,
   );
+  const creatorWeight = ratingCreator
+    ? customerWeight(ratingCreator, roadmap.plannerCustomerWeights)
+    : 0;
 
-  const ratingValue = (rating: Taskrating) => {
-    const ratingCreator = allCustomers.find(
-      ({ id }) => id === rating.forCustomer,
-    );
-
-    const ratingCreatorValue = ratingCreator
-      ? customerWeight(ratingCreator, roadmap.plannerCustomerWeights)
-      : 0;
-
-    let creatorPlannerWeight = roadmap.plannerCustomerWeights?.find(
-      ({ customerId }) => customerId === rating.forCustomer,
-    )?.weight;
-    if (creatorPlannerWeight === undefined) creatorPlannerWeight = 1;
+  return rating.value * creatorWeight;
+};
 
-    let creatorValueWeight = 0;
-    if (ratingCreator === undefined) creatorValueWeight = 1;
+export const taskWeightedValueSummaryInto = (
+  result: RatingsSummary,
+  task: Task,
+  allCustomers: Customer[],
+  roadmap: Roadmap,
+) =>
+  task.ratings
+    .filter(({ dimension }) => dimension === TaskRatingDimension.BusinessValue)
+    .map(ratingValue(allCustomers, roadmap))
+    .reduce((acc, num) => {
+      const previous = acc ?? new RatingsSummary();
+      return previous.add(num);
+    }, result);
```

Comment:
```suggestion
    .reduce((acc, num) => acc.add(num), result);
```

either there is no need for the conditional
or the parameter type is incorrect

Reply:
Thank you for pointing this out: it's now corrected